pkg/cnbutils/copy_project.go

```diff
@@ -11,23 +12,94 @@ import (
 	ignore "github.com/sabhiram/go-gitignore"
 )
 
-func CopyProject(source, target string, include, exclude *ignore.GitIgnore, utils BuildUtils) error {
-	sourceFiles, _ := utils.Glob(path.Join(source, "**"))
+func isFiltered(path string, knownSymlinks []string) bool {
+	for _, symlink := range knownSymlinks {
+		if strings.HasPrefix(path, symlink) {
+			return true
+		}
+	}
+	return false
+}
+
+func filterSourceFiles(sourceFiles []string, utils BuildUtils) ([]string, error) {
+	filteredFiles := []string{}
+	knownSymlinks := []string{}
+
+	for _, sourceFile := range sourceFiles {
+		if isFiltered(sourceFile, knownSymlinks) {
+			continue
+		}
+
+		isSymlink, err := symlinkExists(sourceFile, utils)
+		if err != nil {
+			return nil, err
+		}
+
+		isDir, err := utils.DirExists(sourceFile)
+		if err != nil {
+			return nil, err
+		}
+
+		if isSymlink {
+			if isDir {
+				log.Entry().Debugf("Ignoring any path below %s", sourceFile)
+				knownSymlinks = append(knownSymlinks, sourceFile)
+			}
+			filteredFiles = append(filteredFiles, sourceFile)
+		} else {
+			if !isFiltered(sourceFile, knownSymlinks) {
```

Comment:
Probably I haven't understood the problem as well as I thought, but this `if` confuses me? Is the `!` right? If it is *NOT* prefixed by a symlink, it will be filtered?

I thought it should look something like 
```
if isFiltered(sourceFile, knownSymlinks) {
	filteredFiles = append(filteredFiles, sourceFile)
}
```

and this can be called everytime (no need for a `if`) since a symlink will always return `true`.